### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by gautamxyz**
Completed multipliers experiment

### CODE REVIEW

File: experiment/simulation/js/layout.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  resize();
+}
+
+window.changeTabs = changeTabs;
+
+// Toolbar
+
+function updateToolbar() {
+  let elem = "";
+  elem =
```
Comment (raj-vlabs) [inline experiment/simulation/js/layout.js:89]: Why do we need 2 statements here to assign the value to elem?

File: experiment/simulation/js/layout.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    '<div class="component-button and" onclick="addGate(event)">AND</div><div class="component-button or" onclick="addGate(event)">OR</div><div class="component-button not" onclick="addGate(event)">NOT</div><div class="component-button nand" onclick="addGate(event)">NAND</div><div class="component-button nor" onclick="addGate(event)">NOR</div><div class="component-button xor" onclick="addGate(event)">XOR</div><div class="component-button xnor" onclick="addGate(event)">XNOR</div>'; 
+  document.getElementById("toolbar").innerHTML = elem;
+}
+
+// Clear observations
+export function clearObservations() {
+  document.getElementById("table-body").innerHTML = "";
+  let head = "";
+  head =
```
Comment (raj-vlabs) [inline experiment/simulation/js/layout.js:98]: Why do we need 2 statements here to assign the value to head?

File: experiment/simulation/practice.html [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+						<li class="is-active" id="task1" onclick="changeTabs(event)">
+							<a>
+								Two Bit Multiplier
+							</a>
+						</li>
+					</ul>
+				</div>
+			</div>
+		</div>
+		<div class="columns is-centered is-flex">
```
Comment (raj-vlabs) [inline experiment/simulation/practice.html:32]: Why do we need is-flex with columns?

File: experiment/simulation/practice.html [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+						</ul>
+						<b style="font-size: 1.1rem;">Here C3 is the MSB of the output and thus the leftmost bit in the observed and expected values, C0 is the LSB of the output thus the rightmost bit in observed and expected values, A1 and B1 are MSBs of the
+							2 input numbers and A0 and B0 are LSBs of the 2 input numbers.</b>
+					</div>
+				</div>
+			</div>
+		</div>
+
+	</div>
+	<div class="columns is-centered is-flex is-flex-direction-row is-flex-wrap-wrap is-align-content-start">
```
Comment (raj-vlabs) [inline experiment/simulation/practice.html:56]: Why do we need is-flex with columns?

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
1. jsPlumb should be imported from a cdn.
2. Procedure needs major improvements.

Code looks very similar to the ALU experiment and many review comments are repeated. Please resolve the same review comments for all 4 experiments and submit for review again.

File: experiment/simulation/demo.html [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+                            <path id="path1AND" fill="#29e" stroke="currentColor" stroke-width="3"
+                                d="M 221 400 v 80 h 40 Q 301 440 261 400 h -40" />
+                        </g>
+                        <g id="and">
+                            <path id="path1AND" fill="#29e" stroke="currentColor" stroke-width="3"
+                                d="M 221 500 v 80 h 40 Q 301 540 261 500 h -40" />
+                        </g>
+
+                        <g id="and">
+                            <path id="path1AND" fill="#29e" stroke="currentColor" stroke-width="3"
```
[the remark this answers is not in the document]
Comment (Aditya-debug15) [reply experiment/simulation/demo.html:140]: added 3 classes gate-demo, text-demo and wire-demo.

File: experiment/simulation/js/animation.js [hunk 6]
```diff
@@ -0,0 +1,796 @@
...
+const width = window.innerWidth;
+const instructionBox = document.getElementsByClassName("instructions-box")[0];
+const svg = document.querySelector(".svg");
+const svgns = "http://www.w3.org/2000/svg";
+
+const EMPTY = "";
+const ZERO="rgb(238, 235, 34)"
+const STATUS = document.getElementById("play-or-pause");
+const OBSERV = document.getElementById("observations");
+const SPEED = document.getElementById("speed");
```
[the remark this answers is not in the document]
Comment (Aditya-debug15) [reply experiment/simulation/js/animation.js:29]: changed their naming

File: experiment/simulation/js/animation.js [hunk 7]
```diff
@@ -0,0 +1,796 @@
...
+//execution starts here
+gsap.registerPlugin(MotionPathPlugin);
+demoWidth();
+instructionBoxInit();
+textIOInit();
+outputCoordinates();
+inputDots();
+outputDisappear();
+
+timeline.add(inputTextAppear, 0);
```
[the remark this answers is not in the document]
Comment (gautamxyz) [reply experiment/simulation/js/animation.js:775]: done

File: experiment/pretest.json [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            "question": "3. Underflow is a condition that happens when",
+            "answers": {
+                "a": "overflow is not happening",
+                "b": "the result that is smaller than the smallest number obtained",
+                "c": "We got a NaN on our computer screen",
+                "d": "None of the above"
+            },
+            "correctAnswer": "b",
+            "explanations": {"b":"Overflow is the result that is smaller than the smallest number obtained"},
```
[the remark this answers is not in the document]
Comment (gautamxyz) [reply experiment/pretest.json:37]: done

File: experiment/simulation/js/animation.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (Aditya-debug15) [reply experiment/simulation/js/animation.js:775]: done

File: experiment/simulation/js/layout.js (hunk 1 above)
Comment (gautamxyz) [reply experiment/simulation/js/layout.js:89]: removed the redundant statement

File: experiment/simulation/js/layout.js (hunk 2 above)
Comment (gautamxyz) [reply experiment/simulation/js/layout.js:98]: removed the redundant statement

File: experiment/simulation/practice.html (hunk 4 above)
Comment (gautamxyz) [reply experiment/simulation/practice.html:56]: removed

File: experiment/simulation/js/layout.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+];
+
+// Contextmenu
+const menu = document.querySelector(".menu");
+const menuOption = document.querySelector(".menu-option");
+let menuVisible = false;
+
+const toggleMenu = (command) => {
+  menu.style.display = command === "show" ? "block" : "none";
+  menuVisible = !menuVisible;
```
[the remark this answers is not in the document]
Comment (gautamxyz) [reply experiment/simulation/js/layout.js:33]: There was some bug coming when deriving from visibility property that's why as a workaround we stored it in a variable menu visible which indicates whether it's visible or not.

**Review by raj-vlabs — APPROVED** [review]

**Pull request merged**